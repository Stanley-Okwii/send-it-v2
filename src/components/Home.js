import React, { Component } from "react";
import { Container, Menu } from "semantic-ui-react";

import ParcelTable from "./Table";
import MenuBar from "./MenuBar";
import AddParcel from "../containers/AddParcel";

import "semantic-ui-css/semantic.min.css";

import "../css/App.css";

export class Home extends Component {
  constructor(props) {
    super(props);

    this.state = {
      activeItem: "add",
    };
  }

  handleItemClick = (e, { name }) => this.setState({ activeItem: name });

  render() {
    const { activeItem } = this.state;
    return (
      <Container style={{ marginTop: "3em" }}>
        <MenuBar />
        <Menu secondary pointing>
          <Menu.Item
            name="add"
            active={activeItem === "add"}
            onClick={this.handleItemClick}
          />
          <Menu.Item
            name="View all"
            active={activeItem === "View all"}
            onClick={this.handleItemClick}
          />
        </Menu>
        {this.renderSelection()}
      </Container>
    );
  }

  renderSelection = () => {
    const { activeItem } = this.state;
    if (activeItem === "View all") {
      return <ParcelTable />;
    }
    if (activeItem === "add") {
      return (
        <AddParcel/>
      );
    }
  };
}
export default Home;
