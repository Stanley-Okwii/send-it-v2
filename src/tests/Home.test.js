import React from "react";
import { shallow } from "enzyme";
import Home from "../components/Home";

describe("Home ", () => {
  it(" renders without crashing", () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });

  it(" responds to click events", () => {
    const wrapper = shallow(<Home />);
    const event = {};
    wrapper.instance().handleItemClick(event, { name: "active" });
    expect(wrapper.instance().state.activeItem).toEqual("active");
  });

  it(" displays the view parcel tab", () => {
    const wrapper = shallow(<Home />);
    const event = {};
    wrapper.instance().handleItemClick(event, { name: "all" });
    expect(wrapper.instance().state.activeItem).toEqual("all");
  });
});
