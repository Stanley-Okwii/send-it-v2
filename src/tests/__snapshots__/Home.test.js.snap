// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home   renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <withRouter(MenuBar) />,
        <Menu
          pointing={true}
          secondary={true}
        >
          <MenuItem
            active={true}
            name="add"
            onClick={[Function]}
          />
          <MenuItem
            active={false}
            name="View all"
            onClick={[Function]}
          />
        </Menu>,
        <Connect(AddParcel) />,
      ],
      "style": Object {
        "marginTop": "3em",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <MenuItem
              active={true}
              name="add"
              onClick={[Function]}
            />,
            <MenuItem
              active={false}
              name="View all"
              onClick={[Function]}
            />,
          ],
          "pointing": true,
          "secondary": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": true,
              "name": "add",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "name": "View all",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <withRouter(MenuBar) />,
          <Menu
            pointing={true}
            secondary={true}
          >
            <MenuItem
              active={true}
              name="add"
              onClick={[Function]}
            />
            <MenuItem
              active={false}
              name="View all"
              onClick={[Function]}
            />
          </Menu>,
          <Connect(AddParcel) />,
        ],
        "style": Object {
          "marginTop": "3em",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <MenuItem
                active={true}
                name="add"
                onClick={[Function]}
              />,
              <MenuItem
                active={false}
                name="View all"
                onClick={[Function]}
              />,
            ],
            "pointing": true,
            "secondary": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": true,
                "name": "add",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "name": "View all",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
